endgame_postprocessing/post_processing/iu_data.py

```diff
@@ -0,0 +1,87 @@
+import re
+
+import pandas as pd
+
+from endgame_postprocessing.post_processing.disease import Disease
+
+
+def _is_valid_iu_code(iu_code):
+    return re.match(r"[A-Z]{3}\d{5}$", iu_code)
+
+
+def preprocess_iu_meta_data(input_data: pd.DataFrame):
+    deduped_input_data = input_data.drop_duplicates()
+    new_iu_code = deduped_input_data.ADMIN0ISO3 + deduped_input_data["IU_ID"].apply(
+        lambda id: str.zfill(str(id), 5)
+    )
+    deduped_input_data.loc[:, "IU_CODE"] = new_iu_code
+    return deduped_input_data
+
+
+class IUData:
+
+    def __init__(self, input_data: pd.DataFrame, disease: Disease):
+        self.disease = disease
+        # TODO: validate the required columns are as expcted
+
+        population_column_name = self._get_priority_population_column_name()
+        if population_column_name not in input_data.columns:
+            raise InvalidIUDataFile(
+                f"No priority population found for disease {self.disease.name}"
+                f", expected {population_column_name}"
+            )
+
+        if input_data["IU_CODE"].nunique() != len(input_data):
+            raise InvalidIUDataFile("Duplicate IUs found")
+
+        self.input_data = input_data.drop_duplicates()
```

Comment:
:pick: If you're going to raise an error for duplicate IU Codes, then I think a call to drop_duplicates is redundant

Reply:
Good spot, thank you 💚